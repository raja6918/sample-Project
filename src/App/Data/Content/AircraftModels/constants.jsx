export const orderBy = 'code';
export const order = 'inc';
export const endpoint = '/data/aircrafts/models';
export const MODEL_NAME_MAX_LENGTH = 50;
export const MODEL_CODE_MAX_LENGTH = 10;
export const MODEL_NAME_REGEX = '^[a-zA-Z0-9].*';
export const MODEL_CODE_REGEX = `^[a-zA-Z0-9]{1,${MODEL_CODE_MAX_LENGTH}}$`;

export const INITIAL_ITEMS = 50;

export const type = 'DATA.aircraftModels.type';

export const getHeaders = t => [
  {
    field: 'code',
    displayName: t('DATA.aircraftModels.headers.model'),
  },
  {
    field: 'name',
    displayName: t('DATA.aircraftModels.headers.name'),
  },
];
