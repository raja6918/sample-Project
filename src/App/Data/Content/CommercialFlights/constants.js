import { evaluateRegex } from '../../utils/utils.js';
import {
  operationDaysTransformer,
  dateFormatTransformer,
} from './../../../../components/GenericTable/transformers';
import DateRangePicker from '../../../../components/DateRangePicker';
import TimeRangePicker from '../../../../components/TimeRangePicker';

export const translationPath = 'DATA.commercialFlights';
export const COMMERCIAL_FLIGHTS_FORM = 'DATA.commercialFlights.form';
export const COMMERCIAL_FLIGHTS_GENERAL = `${COMMERCIAL_FLIGHTS_FORM}.section.general`;
export const COMMERCIAL_FLIGHTS_CREW = `${COMMERCIAL_FLIGHTS_FORM}.section.crew`;
export const COMMERCIAL_FLIGHTS_TIME = `${COMMERCIAL_FLIGHTS_FORM}.section.extraTime`;
export const COMMERCIAL_FLIGHTS_INFO = `${COMMERCIAL_FLIGHTS_FORM}.section.extraInformation`;
export const COMMERCIAL_FLIGHTS_ERRORS = 'ERRORS.COMMERCIAL_FLIGHTS';

export const POSITIVE_INTEGERS_REGEX = /^\d+$/g;
export const NAME_REGEX = '^[a-zA-Z0-9].*';
export const CHAR_REGEX = '^[a-zA-Z].*';
export const FLIGHT_DESIGNATOR_REGEX = '^[a-zA-Z]{2,3}[0-9]{1,4}$';
export const VALID_STRING_REGEX = '^[a-zA-Z0-9]+?(.)*?$';
export const AIRLINE_CODE_REGEX = '^[a-zA-Z]{2,3}$';
export const AIRCRAFTTYPE_CODE_REGEX = '^[a-zA-Z0-9]{3,4}$';
export const CABIN_CONFIG_REGEX = '^[a-zA-Z0-9]+$';

export const orderBy = 'flightDesignator';
export const order = 'inc';

const UTC = 'UTC';
export const TIME_REFERENCE = UTC;
export const INITIAL_ITEMS = 50;

export const type = 'DATA.commercialFlights.type';

export const getHeaders = (t, scenarioStartDate) => [
  {
    field: 'flightDesignator',
    displayName: t('DATA.commercialFlights.headers.flightDesignator'),
    disableFilter: false,
  },
  {
    field: 'departureStationCode',
    displayName: t('DATA.commercialFlights.headers.departureStationCode'),
    disableFilter: false,
  },
  {
    field: 'arrivalStationCode',
    displayName: t('DATA.commercialFlights.headers.arrivalStationCode'),
    disableFilter: false,
  },
  {
    field: 'startTime',
    displayName: t('DATA.commercialFlights.headers.startTime'),
    disableFilter: false,
    component: TimeRangePicker,
    filterTransformer: date => date.replace(' - ', '-'),
  },
  {
    field: 'endTime',
    displayName: t('DATA.commercialFlights.headers.endTime'),
    disableFilter: false,
    component: TimeRangePicker,
    filterTransformer: date => date.replace(' - ', '-'),
  },
  {
    field: 'operationDays',
    displayName: t('DATA.commercialFlights.headers.operationDays'),
    transformer: operationDaysTransformer,
    disableSort: true,
    disableFilter: true,
  },
  {
    field: 'firstDeparture',
    displayName: t('DATA.commercialFlights.headers.firstDepartureDate'),
    disableFilter: false,
    filterType: 'YYYY-MM-DD',
    component: DateRangePicker,
    componentProps: { startupDate: scenarioStartDate },
    filterTransformer: date => date.replace(' - ', ':'),
    transformer: dateFormatTransformer,
  },
  {
    field: 'lastDeparture',
    displayName: t('DATA.commercialFlights.headers.lastDepartureDate'),
    disableFilter: false,
    filterType: 'YYYY-MM-DD',
    component: DateRangePicker,
    componentProps: { startupDate: scenarioStartDate },
    filterTransformer: date => date.replace(' - ', ':'),
    transformer: dateFormatTransformer,
  },
  {
    field: 'aircraftType',
    displayName: t('DATA.commercialFlights.headers.aircraftTypeCode'),
    disableFilter: false,
  },
  {
    field: 'aircraftConfigurationVersion',
    displayName: t(
      'DATA.commercialFlights.headers.aircraftConfigurationVersion'
    ),
    disableFilter: false,
  },
];

export const COMMERCIAL_FLIGHTS_FIELDS = [
  'flightNumber',
  'airlineCode',
  'departureStationCode',
  'arrivalStationCode',
  'aircraftType',
  'operationalSuffix',
  'passengerTerminalDeparture',
  'passengerTerminalArrival',
  'aircraftConfigurationVersion',
  'tags',
  'startTime',
  'endTime',
  'flightDesignator',
  'startDates',
  'flightInstances',
];

export const REQUIRED_FIELDS = [
  'flightNumber',
  'airlineCode',
  'departureStationCode',
  'arrivalStationCode',
  'startTime',
  'endTime',
  'aircraftType',
  'startDates',
];

export const isRequiredPredicate = val => val !== '' && val !== null;

const isIntegerGreaterThanOrEqualToZeroPredicate = val =>
  evaluateRegex(POSITIVE_INTEGERS_REGEX, val) &&
  Number(val) >= 0 &&
  Number(val) <= 9999;

const isNotEmptyArray = val => val.length;

export const customPredicates = {
  airlineCode: val => evaluateRegex(AIRLINE_CODE_REGEX, val),
  aircraftType: val => evaluateRegex(AIRCRAFTTYPE_CODE_REGEX, val),
  flightNumber: isIntegerGreaterThanOrEqualToZeroPredicate,
  operationalSuffix: val => evaluateRegex(CHAR_REGEX, val),
  deadheadSeatsNumber: isIntegerGreaterThanOrEqualToZeroPredicate,
  onwardFlightDayOffset: isIntegerGreaterThanOrEqualToZeroPredicate,
  extraBriefingCabin: isIntegerGreaterThanOrEqualToZeroPredicate,
  extraBriefingFlightDeck: isIntegerGreaterThanOrEqualToZeroPredicate,
  extraDebriefingCabin: isIntegerGreaterThanOrEqualToZeroPredicate,
  extraDebriefingFlightDeck: isIntegerGreaterThanOrEqualToZeroPredicate,
  onwardFlightDesignator: val => evaluateRegex(FLIGHT_DESIGNATOR_REGEX, val),
  tailNumber: val => evaluateRegex(VALID_STRING_REGEX, val),
  aircraftConfigurationVersion: val => evaluateRegex(CABIN_CONFIG_REGEX, val),
  startDates: isNotEmptyArray,
};
