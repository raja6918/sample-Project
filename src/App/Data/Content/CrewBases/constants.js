export const orderBy = 'name';
export const order = 'inc';

export const type = 'DATA.crewBases.type';

export const INITIAL_ITEMS = 100;

export const getHeaders = t => [
  {
    field: 'name',
    displayName: t('DATA.crewBases.headers.name'),
  },
  {
    field: 'countryDisplayName',
    displayName: t('DATA.crewBases.headers.country'),
  },
  {
    field: 'stations',
    displayName: t('DATA.crewBases.headers.stations'),
    sortCriteria: 'stationCodes',
  },
];
