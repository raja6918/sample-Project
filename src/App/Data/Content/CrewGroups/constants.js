import { ellipsisTransformer } from './../../../../components/GenericTable/transformers';
import { evaluateRegex } from '../../utils/utils.js';

export const orderBy = 'name';
export const order = 'inc';

export const type = 'DATA.crewGroups.type';

export const INITIAL_ITEMS = 150;

export const NAME_REGEX = '^[a-zA-Z0-9].*';

export const CREWGROUPS_FORM = 'DATA.crewGroups.form';
export const CREWGROUPS_GENERAL = `${CREWGROUPS_FORM}.section.general`;
export const CREWGROUPS_ERRORS = 'ERRORS.CREW_GROUPS';

export const CREW_GROUPS_FIELDS = [
  'name',
  'positionCodes',
  'airlineCodes',
  'aircraftTypeCodes',
  'ruleset',
];
export const REQUIRED_FIELDS = [
  'name',
  'positionCodes',
  'airlineCodes',
  'aircraftTypeCodes',
  'ruleset',
];

export const getHeaders = t => [
  {
    field: 'name',
    displayName: t('DATA.crewGroups.headers.name'),
  },
  {
    field: 'positions',
    displayName: t('DATA.crewGroups.headers.positions'),
    transformer: ellipsisTransformer,
    sortCriteria: 'positionCodes',
  },
  {
    field: 'airlines',
    displayName: t('DATA.crewGroups.headers.airlines'),
    sortCriteria: 'airlineCodes',
  },
  {
    field: 'aircraftTypes',
    displayName: t('DATA.crewGroups.headers.aircraftTypes'),
    transformer: ellipsisTransformer,
    sortCriteria: 'aircraftTypeCodes',
  },
  {
    field: 'rulesetDisplayName',
    displayName: t('DATA.crewGroups.headers.ruleset'),
    transformer: ellipsisTransformer,
  },
];

export const isRequiredPredicate = val => val !== '' && val !== null;
const arrayIsNotEmpty = arr => Array.isArray(arr) && arr.length > 0;

export const customPredicates = {
  name: val => evaluateRegex(NAME_REGEX, val),
  positionCodes: arrayIsNotEmpty,
  airlineCodes: arrayIsNotEmpty,
  aircraftTypeCodes: arrayIsNotEmpty,
};
