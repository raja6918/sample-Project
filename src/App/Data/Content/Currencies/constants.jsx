export const orderBy = 'code';
export const order = 'inc';

export const CURRENCY_CODE_MAX_LENGTH = 3;
export const CURRENCY_NAME_MAX_LENGTH = 50;
export const CURRENCY_EXCHANGE_RATE = 50;
export const CURRENCY_CODE_REGEX = `[A-Za-z]{${CURRENCY_CODE_MAX_LENGTH}}`;
export const CURRENCY_NAME_REGEX = '^[a-zA-Z0-9].*';
export const CURRENCY_RATE_REGEX = /^(?=.*[1-9])\d{1,6}(?:\.\d{1,6})?\s*$/g;

export const INITIAL_ITEMS = 250;

export const type = 'Currency';

export const getHeaders = t => [
  {
    field: 'code',
    displayName: t('DATA.currencies.headers.code'),
  },
  {
    field: 'name',
    displayName: t('DATA.currencies.headers.name'),
  },
  {
    field: 'exchangeRate',
    displayName: t('DATA.currencies.headers.exchangeRate'),
  },
];
