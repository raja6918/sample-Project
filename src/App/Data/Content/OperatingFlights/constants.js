import { evaluateRegex } from '../../utils/utils.js';
import {
  operationDaysTransformer,
  crewCompositionTransformer,
  dateFormatTransformer,
} from './../../../../components/GenericTable/transformers';
import DateRangePicker from '../../../../components/DateRangePicker';
import TimeRangePicker from '../../../../components/TimeRangePicker';

export const translationPath = 'DATA.operatingFlights';
export const OPERATING_FLIGHTS_FORM = 'DATA.operatingFlights.form';
export const OPERATING_FLIGHTS_GENERAL = `${OPERATING_FLIGHTS_FORM}.section.general`;
export const OPERATING_FLIGHTS_CREW = `${OPERATING_FLIGHTS_FORM}.section.crew`;
export const OPERATING_FLIGHTS_TIME = `${OPERATING_FLIGHTS_FORM}.section.extraTime`;
export const OPERATING_FLIGHTS_INFO = `${OPERATING_FLIGHTS_FORM}.section.extraInformation`;
export const OPERATING_FLIGHTS_ERRORS = 'ERRORS.OPERATING_FLIGHTS';

export const POSITIVE_INTEGERS_REGEX = /^\d+$/g;
export const NAME_REGEX = '^[a-zA-Z0-9].*';
export const CHAR_REGEX = '^[a-zA-Z].*';
export const FLIGHT_DESIGNATOR_REGEX = '^[a-zA-Z]{2,3}[0-9]{1,4}$';
export const VALID_STRING_REGEX = '^[a-zA-Z0-9]+?(.)*?$';
export const CABIN_CONFIG_REGEX = '^[a-zA-Z0-9]+$';

export const orderBy = 'flightDesignator';
export const order = 'inc';

const UTC = 'UTC';
export const TIME_REFERENCE = UTC;
export const INITIAL_ITEMS = 50;

export const type = 'DATA.operatingFlights.type';

export const getHeaders = (t, scenarioStartDate) => [
  {
    field: 'flightDesignator',
    displayName: t('DATA.operatingFlights.headers.flightDesignator'),
    disableFilter: false,
  },
  {
    field: 'departureStationCode',
    displayName: t('DATA.operatingFlights.headers.departureStationCode'),
    disableFilter: false,
  },
  {
    field: 'arrivalStationCode',
    displayName: t('DATA.operatingFlights.headers.arrivalStationCode'),
    disableFilter: false,
  },
  {
    field: 'startTime',
    displayName: t('DATA.operatingFlights.headers.startTime'),
    disableFilter: false,
    component: TimeRangePicker,
    filterTransformer: date => date.replace(' - ', '-'),
  },
  {
    field: 'endTime',
    displayName: t('DATA.operatingFlights.headers.endTime'),
    disableFilter: false,
    component: TimeRangePicker,
    filterTransformer: date => date.replace(' - ', '-'),
  },
  {
    field: 'operationDays',
    displayName: t('DATA.operatingFlights.headers.operationDays'),
    transformer: operationDaysTransformer,
    disableSort: true,
    disableFilter: true,
  },
  {
    field: 'firstDepartureDate',
    displayName: t('DATA.operatingFlights.headers.firstDepartureDate'),
    disableFilter: false,
    filterType: 'YYYY-MM-DD',
    component: DateRangePicker,
    componentProps: { startupDate: scenarioStartDate },
    filterTransformer: date => date.replace(' - ', ':'),
    transformer: dateFormatTransformer,
  },
  {
    field: 'lastDepartureDate',
    displayName: t('DATA.operatingFlights.headers.lastDepartureDate'),
    disableFilter: false,
    filterType: 'YYYY-MM-DD',
    component: DateRangePicker,
    componentProps: { startupDate: scenarioStartDate },
    filterTransformer: date => date.replace(' - ', ':'),
    transformer: dateFormatTransformer,
  },
  {
    field: 'onwardFlightDesignatorWithOffset',
    displayName: t('DATA.operatingFlights.headers.onwardFlightDesignator'),
    sortCriteria: ['onwardFlightDesignator', 'onwardFlightDayOffset'],
    disableFilter: false,
    filterName: 'onwardFlightDesignator',
  },
  {
    field: 'aircraftTypeCode',
    displayName: t('DATA.operatingFlights.headers.aircraftTypeCode'),
    disableFilter: false,
  },
  {
    field: 'crewComposition',
    displayName: t('DATA.operatingFlights.headers.crewComposition'),
    transformer: crewCompositionTransformer,
    disableSort: true,
    disableFilter: true,
  },
];

export const OPERATING_FLIGHTS_FIELDS = [
  'flightNumber',
  'airlineCode',
  'departureStationCode',
  'arrivalStationCode',
  'tailNumber',
  'aircraftTypeCode',
  'duration',
  'sequenceNumber',
  'serviceTypeCode',
  'onwardFlightDesignator',
  'onwardFlightDayOffset',
  'operationalSuffix',
  'extraBriefingFlightDeck',
  'extraBriefingCabin',
  'extraDebriefingFlightDeck',
  'extraDebriefingCabin',
  'passengerTerminalDeparture',
  'passengerTerminalArrival',
  'aircraftConfigurationVersion',
  'tags',
  'crewComposition',
  'startTime',
  'endTime',
  'flightDesignator',
  'flightInstances',
  'deadheadSeatsNumber',
  'startDates',
];

export const REQUIRED_FIELDS = [
  'flightNumber',
  'airlineCode',
  'departureStationCode',
  'arrivalStationCode',
  'startTime',
  'endTime',
  'aircraftTypeCode',
  'startDates',
];

export const isRequiredPredicate = val => val !== '' && val !== null;

const isIntegerGreaterThanOrEqualToZeroPredicate = val =>
  evaluateRegex(POSITIVE_INTEGERS_REGEX, val) &&
  Number(val) >= 0 &&
  Number(val) <= 9999;

const isNotEmptyArray = val => val.length;

export const customPredicates = {
  flightNumber: isIntegerGreaterThanOrEqualToZeroPredicate,
  operationalSuffix: val => evaluateRegex(CHAR_REGEX, val),
  deadheadSeatsNumber: isIntegerGreaterThanOrEqualToZeroPredicate,
  onwardFlightDayOffset: isIntegerGreaterThanOrEqualToZeroPredicate,
  extraBriefingCabin: isIntegerGreaterThanOrEqualToZeroPredicate,
  extraBriefingFlightDeck: isIntegerGreaterThanOrEqualToZeroPredicate,
  extraDebriefingCabin: isIntegerGreaterThanOrEqualToZeroPredicate,
  extraDebriefingFlightDeck: isIntegerGreaterThanOrEqualToZeroPredicate,
  onwardFlightDesignator: val => evaluateRegex(FLIGHT_DESIGNATOR_REGEX, val),
  tailNumber: val => evaluateRegex(VALID_STRING_REGEX, val),
  aircraftConfigurationVersion: val => evaluateRegex(CABIN_CONFIG_REGEX, val),
  startDates: isNotEmptyArray,
};
