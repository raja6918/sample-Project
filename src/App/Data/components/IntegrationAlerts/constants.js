export const ALERT_PAGE_SIZE = 20;
export const SCROLL_LIMIT = 50;
