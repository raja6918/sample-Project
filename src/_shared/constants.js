export const DEBUG_MODE = process.env.NODE_ENV === 'production' ? false : true;
