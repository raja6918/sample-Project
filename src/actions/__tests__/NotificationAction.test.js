import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import * as notificationService from '../../services/Notifications';
import {
  notificationIncrement,
  notificationDecrement,
  addNotificationsToStore,
  viewedAllStatus,
  notificationCleanUp,
  closeNotificationBar,
  addNewNotificationToStore,
  fetchingNotification,
  deleteSingleNotification,
  deleteAllNotification,
  updateNotificationAsSeen,
  fetchNotifications,
} from '../generic';
import { GENERAL } from '../../constants';

const newNotification = {
  data: [
    {
      activeTime: '0:00:10',
      creationDateTime: '2020-08-12T07:57:15+0000',
      elapsedTime: '0:00:15',
      jobId: '139288-data-test',
      eventId: '139288-data-test',
      lastModified: '2020-08-12T07:57:20+0000',
      message: null,
      status: 'Done-success',
      alertType: 'SUCCESS',
      solverName: 'data test',
      scenarioId: '3',
      scenarioName: 'Jazz Template with Pairings',
      jobOwner: '5',
      state: 'Done-success',
      eventType: 'solver',
      solverId: '46',
      notificationType: 'solver',
    },
    {
      activeTime: '0:00:10',
      creationDateTime: '2020-08-12T07:57:15+0000',
      elapsedTime: '0:00:15',
      jobId: '139288-data-test-2',
      eventId: '139288-data-test-2',
      lastModified: '2020-08-12T07:57:20+0000',
      message: null,
      status: 'Done-failed',
      alertType: 'ERROR',
      solverName: 'data test 2',
      scenarioId: '3',
      scenarioName: 'Test',
      jobOwner: '5',
      state: 'Done-failed',
      eventType: 'solver',
      solverId: '48',
      notificationType: 'solver',
    },
  ],
  totalCount: 2,
  startIndex: 0,
  notViewedCount: 2,
};

notificationService.getNotifications = jest.fn(() =>
  Promise.resolve(newNotification)
);
const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const defaultState = {
  count: 0,
  temp: [],
  notifications: [],
  totalSize: 0,
  isFetching: false,
  start: 0,
  viewedAll: false,
  closeNavBar: false,
};
const store = mockStore(defaultState);

describe(`Tests related to Notification Actions`, () => {
  beforeEach(() => {
    store.clearActions();
  });

  test(`check notification increment action gets dispatched as expected`, () => {
    const expectedResult = [
      {
        type: GENERAL.NOTIFICATION_INCREMENT,
        payload: 1,
      },
    ];
    store.dispatch(notificationIncrement(1));
    expect(store.getActions()).toEqual(expectedResult);
    expect(store.getActions()).toMatchSnapshot();
  });

  test('check notification reset works as expected', () => {
    const expectedResult = [
      {
        type: GENERAL.NOTIFICATION_RESET,
        payload: 0,
      },
    ];
    store.dispatch(notificationDecrement(0));
    expect(store.getActions()).toEqual(expectedResult);
    expect(store.getActions()).toMatchSnapshot();
  });

  test(`check adding a notification action is dispatched properly`, () => {
    const expectedResult = [
      {
        type: GENERAL.NOTIFICATION_SUCCESS,
        payload: newNotification.data[0],
      },
    ];
    store.dispatch(addNotificationsToStore(newNotification.data[0]));
    expect(store.getActions()).toEqual(expectedResult);
    expect(store.getActions()).toMatchSnapshot();
  });

  test(`check the view all action is dispatched`, () => {
    const expectedResult = [
      {
        type: GENERAL.VIEWED_ALL_NOTIFICATION,
      },
    ];
    store.dispatch(viewedAllStatus());
    expect(store.getActions()).toEqual(expectedResult);
    expect(store.getActions()).toMatchSnapshot();
  });
  test(`check the clean up action is dispatched`, () => {
    const expectedResult = [
      {
        type: GENERAL.NOTIFICATION_CLEANUP,
      },
    ];
    store.dispatch(notificationCleanUp());
    expect(store.getActions()).toEqual(expectedResult);
    expect(store.getActions()).toMatchSnapshot();
  });

  test(`check the notification Bar action is dispatched`, () => {
    const expectedResult = [
      {
        type: GENERAL.NOTIFICATION_BAR_CLOSE,
        bool: true,
      },
    ];
    store.dispatch(closeNotificationBar(true));
    expect(store.getActions()).toEqual(expectedResult);
    expect(store.getActions()).toMatchSnapshot();
  });

  test(`check whether the single notificationw action is dispatched`, () => {
    const expectedResult = [
      {
        type: GENERAL.NOTIFICATION_ADD,
        payload: newNotification.data[0],
      },
    ];
    store.dispatch(addNewNotificationToStore(newNotification.data[0]));
    expect(store.getActions()).toEqual(expectedResult);
    expect(store.getActions()).toMatchSnapshot();
  });

  test(`check whether the fetching notification action is dispatched`, () => {
    const expectedResult = [
      {
        type: GENERAL.NOTIFICATION_FETCHING,
        bool: true,
      },
    ];
    store.dispatch(fetchingNotification(true));
    expect(store.getActions()).toEqual(expectedResult);
    expect(store.getActions()).toMatchSnapshot();
  });

  test(`check whether the single delete notification action is dispatched`, () => {
    const expectedResult = [
      {
        type: GENERAL.NOTIFICATION_SINGLE_DELETE,
        notificationId: 1,
      },
    ];
    store.dispatch(deleteSingleNotification(1));
    expect(store.getActions()).toEqual(expectedResult);
    expect(store.getActions()).toMatchSnapshot();
  });

  test(`check whether the add delete notification action is dispatched`, () => {
    const expectedResult = [
      {
        type: GENERAL.NOTIFICATION_ALL_DELETE,
      },
    ];
    store.dispatch(deleteAllNotification());
    expect(store.getActions()).toEqual(expectedResult);
    expect(store.getActions()).toMatchSnapshot();
  });

  test(`check whether the update all notification action is dispatched`, () => {
    const expectedResult = [
      {
        type: GENERAL.NOTIFICATION_ALL_SEEN,
      },
    ];
    store.dispatch(updateNotificationAsSeen());
    expect(store.getActions()).toEqual(expectedResult);
    expect(store.getActions()).toMatchSnapshot();
  });

  test(`check whether the fetch data action without scroll is dispatched`, () => {
    const expectedResult = [
      {
        type: GENERAL.NOTIFICATION_FETCHING,
        bool: true,
      },
    ];

    store.dispatch(fetchNotifications(1)).then(() => {
      expect(store.getActions()[0]).toEqual(expectedResult);
      expect(store.getActions()[1]).toEqual([
        { ...expectedResult[0], bool: false },
      ]);
    });

    expect(store.getActions()).toMatchSnapshot();
  });

  test(`check whether the fetch action works correctly with scroll`, () => {
    const expectedResult = [
      {
        type: GENERAL.NOTIFICATION_FETCHING,
        bool: true,
      },
    ];

    store.dispatch(fetchNotifications(1, true)).then(() => {
      expect(store.getActions()[0]).toEqual(expectedResult);
      expect(store.getActions()[1]).toEqual([
        { ...expectedResult[0], bool: false },
      ]);
    });

    expect(store.getActions()).toMatchSnapshot();
  });
});
