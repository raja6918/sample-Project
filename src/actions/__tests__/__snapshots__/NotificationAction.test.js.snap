// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests related to Notification Actions check adding a notification action is dispatched properly 1`] = `
Array [
  Object {
    "payload": Object {
      "activeTime": "0:00:10",
      "alertType": "SUCCESS",
      "creationDateTime": "2020-08-12T07:57:15+0000",
      "elapsedTime": "0:00:15",
      "eventId": "139288-data-test",
      "eventType": "solver",
      "jobId": "139288-data-test",
      "jobOwner": "5",
      "lastModified": "2020-08-12T07:57:20+0000",
      "message": null,
      "notificationType": "solver",
      "scenarioId": "3",
      "scenarioName": "Jazz Template with Pairings",
      "solverId": "46",
      "solverName": "data test",
      "state": "Done-success",
      "status": "Done-success",
    },
    "type": "NOTIFICATION_SUCCESS",
  },
]
`;

exports[`Tests related to Notification Actions check notification increment action gets dispatched as expected 1`] = `
Array [
  Object {
    "payload": 1,
    "type": "NOTIFICATION_INCREMENT",
  },
]
`;

exports[`Tests related to Notification Actions check notification reset works as expected 1`] = `
Array [
  Object {
    "payload": 0,
    "type": "NOTIFICATION_RESET",
  },
]
`;

exports[`Tests related to Notification Actions check the clean up action is dispatched 1`] = `
Array [
  Object {
    "type": "NOTIFICATION_CLEANUP",
  },
]
`;

exports[`Tests related to Notification Actions check the notification Bar action is dispatched 1`] = `
Array [
  Object {
    "bool": true,
    "type": "NOTIFICATION_BAR_CLOSE",
  },
]
`;

exports[`Tests related to Notification Actions check the view all action is dispatched 1`] = `
Array [
  Object {
    "type": "VIEWED_ALL_NOTIFICATION",
  },
]
`;

exports[`Tests related to Notification Actions check whether the add delete notification action is dispatched 1`] = `
Array [
  Object {
    "type": "NOTIFICATION_ALL_DELETE",
  },
]
`;

exports[`Tests related to Notification Actions check whether the fetch action works correctly with scroll 1`] = `
Array [
  Object {
    "bool": true,
    "type": "NOTIFICATION_FETCHING",
  },
]
`;

exports[`Tests related to Notification Actions check whether the fetch data action without scroll is dispatched 1`] = `
Array [
  Object {
    "bool": true,
    "type": "NOTIFICATION_FETCHING",
  },
]
`;

exports[`Tests related to Notification Actions check whether the fetching notification action is dispatched 1`] = `
Array [
  Object {
    "bool": true,
    "type": "NOTIFICATION_FETCHING",
  },
]
`;

exports[`Tests related to Notification Actions check whether the single delete notification action is dispatched 1`] = `
Array [
  Object {
    "notificationId": 1,
    "type": "NOTIFICATION_SINGLE_DELTE",
  },
]
`;

exports[`Tests related to Notification Actions check whether the single notificationw action is dispatched 1`] = `
Array [
  Object {
    "payload": Object {
      "activeTime": "0:00:10",
      "alertType": "SUCCESS",
      "creationDateTime": "2020-08-12T07:57:15+0000",
      "elapsedTime": "0:00:15",
      "eventId": "139288-data-test",
      "eventType": "solver",
      "jobId": "139288-data-test",
      "jobOwner": "5",
      "lastModified": "2020-08-12T07:57:20+0000",
      "message": null,
      "notificationType": "solver",
      "scenarioId": "3",
      "scenarioName": "Jazz Template with Pairings",
      "solverId": "46",
      "solverName": "data test",
      "state": "Done-success",
      "status": "Done-success",
    },
    "type": "NOTIFICATION_ADD",
  },
]
`;

exports[`Tests related to Notification Actions check whether the update all notification action is dispatched 1`] = `
Array [
  Object {
    "type": "NOTIFICATION_ALL_SEEN",
  },
]
`;
